src/rgw/rgw_sal_motr.cc

```diff
@@ -1340,53 +1361,31 @@ int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx,
 int MotrObject::get_obj_attrs(RGWObjectCtx* rctx, optional_yield y, const DoutPrefixProvider* dpp, rgw_obj* target_obj)
 {
   if (this->category == RGWObjCategory::MultiMeta)
-    return 0;
-
-  bufferlist bl;
-  string bname, key;
-  int r = read_obj_attrs(dpp, bname, key, bl, target_obj);
-  if (r < 0) {
-    ldpp_dout(dpp, 0) <<__func__<< ": Failed to read bucket name and key. rc=" << r << dendl;
-    return r;
-  }
+   return 0;
 
   rgw_bucket_dir_entry ent;
-  bufferlist& blr = bl;
-  auto iter = blr.cbegin();
-  ent.decode(iter);
-  decode(attrs, iter);
-
-  return 0;
-}
-
-void MotrObject::read_bucket_info(const DoutPrefixProvider* dpp, std::string& bname, std::string& key, rgw_obj* target_obj)
-{
-  if (target_obj) {
-    bname = get_bucket_name(target_obj->bucket.tenant, target_obj->bucket.name);
-    key   = target_obj->key.to_str();
-  } else {
-    bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
-    key   = this->get_key().to_str();
-  }
-  ldpp_dout(dpp, 20) << __func__ << ": for bucket " << bname << "/" << key << dendl;
-}
+  string bname, key;
+  fetch_obj_entry_and_key(dpp, ent, bname, key, target_obj);
 
-int MotrObject::read_obj_attrs(const DoutPrefixProvider* dpp, std::string& bname, std::string& key, bufferlist& bl, rgw_obj* target_obj)
```

Comment:
same here, lets keep this function separate

Reply:
we do not need this function instead we are using get_bucket_dir_ent()

Reply:
alright, didn't know we had a function with similar functionality already